src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -122,8 +122,16 @@ public static class TaskExtensions {
         /// within the timeout specified.
         /// </summary>
         public static T WaitTimeout<T>(this Task<T> task, int msTimeout) {
-            task.Wait(msTimeout);
-            return task.IsCompleted ? task.Result : default(T);
+            bool result = false;
+            try {
+                task.Wait(msTimeout);
+                result = task.IsCompleted;
+            } catch(AggregateException) { }
```

Comment:
So when task completes then ```return task.WaitAndUnwrapExceptions()``` will do that. Now, if ```Wait()``` does throw aggregate, do we want to rethrow? In most cases I think code that calls timeout wait typically only cares if result is valid or a timeout happened.